### PR TITLE
DTSPB-3959 Caveat payment Journey PP changes

### PR DESCRIPTION
**Pull request by IswaryaPepakayala**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-3959

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public static final String PAYMENT_CONFIRMATION = "paymentAcknowledgement";
+
+    private final BusinessValidationMessageRetriever businessValidationMessageRetriever;
+
+    public void validate(CaveatDetails caseDetails) {
+        CaveatData caveatData = caseDetails.getData();
+        List<String> paymentConfirmation = caveatData.getPaymentConfirmCheckbox();
+        String confirmation = (null != paymentConfirmation && !paymentConfirmation.isEmpty())
+                ? caveatData.getPaymentConfirmCheckbox().get(0) : "";
+        log.info("Payment confirmation: {}", confirmation);
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java]: think we can remove this log, as the value is either paymentAcknowledgement or empty string

File: src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if (!PAYMENT_CONFIRMATION.equals(confirmation)) {
+            String userMessage = businessValidationMessageRetriever
+                    .getMessage(PAYMENT_ACKNOWLEDGEMENT, null, Locale.UK);
+            throw new BusinessValidationException(userMessage,
+                    "You must confirm that you understand that you must complete payment after you've submitted your "
+                            + "application");
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java]: this message will be in the log only, don't need to be the same as the one in UI
maybe log the case id here

File: src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java (hunk 1 above)
Comment (IswaryaPepakayala) [reply src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java]: Sorry kept this for testing and forgot to remove it.Removed now

File: src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java (hunk 2 above)
Comment (IswaryaPepakayala) [reply src/main/java/uk/gov/hmcts/probate/validator/CaveatAcknowledgementValidationRule.java]: Changed it

**Review by FeliTam — DISMISSED** [review]

**Review by MarcusHMCTS — APPROVED** [review]

**Pull request merged**